components/tools/FiltersDropdown.js

```diff
@@ -0,0 +1,28 @@
+import { twMerge } from "tailwind-merge";
+
+export default function FiltersDropdown({dataList=[], checkedOptions=[], setStateFunction, className=''}) {
+
+  const handleClickOption = (e, data) => {
+    let tempValueArray = [...checkedOptions]
+    let index = checkedOptions.indexOf(data)
+    if(index>-1){
+      tempValueArray.splice(index, 1);
+    }else{
+      tempValueArray.push(data);
+    }
+    setStateFunction(tempValueArray)
+    e.stopPropagation();
```

Comment:
`e.stopPropagation();` should be fired at the beginning of function. Of course it's not a "wrong", rather a "bad" decision.